api/v1/controllers/samples.js

```diff
@@ -23,6 +23,45 @@ const doPut = require('../helpers/verbs/doPut');
 const u = require('../helpers/verbs/utils');
 const httpStatus = require('../constants').httpStatus;
 const logAPI = require('../../../utils/loggingUtil').logAPI;
+const WorkerLog = require('../../../utils/activityLogs/workerLog');
+const jwtUtil = require('../../../utils/jwtUtil');
+
+/**
+ * Logs worker activity
+ * @param  {Object} reqNext - The request object and The next middleware
+ * function in the stack
+ * @param  {Object} jobDetails - Worker job and job type
+ * @param  {Integer} reqStartTime - Time in ms when req is received
+ */
+function logWorkerActivity(reqNext, jobDetails, reqStartTime) {
+  // create worker activity log object
+  const workerLogObj = new WorkerLog(jobDetails.jobType.BULKUPSERTSAMPLES);
+  u.getUserNameFromToken(reqNext.req, true)
+  .then((username) => {
+    if (username) {
+      workerLogObj.user = username;
+    }
+  })
+  .then(() => jwtUtil.getTokenNameFromToken(reqNext.req))
+  .then((tokenName) => {
+    workerLogObj.token = tokenName;
+  })
+  .then(() => {
+    workerLogObj.ipAddress = u.getIPAddrFromReq(reqNext.req);
```

Comment:
seems like this line 50 can be inside of the previous "then" block since line 47 doesn't return a promise which this would depend on